ydb/core/kqp/ut/view/view_ut.cpp

```diff
@@ -428,6 +447,32 @@ Y_UNIT_TEST_SUITE(TCreateAndDropViewTest) {
         }
     }
 
+    Y_UNIT_TEST(DropViewIfExists) {
+        TKikimrRunner kikimr(TKikimrSettings().SetWithSampleTables(false));
+        EnableViewsFeatureFlag(kikimr);
+        auto session = kikimr.GetQueryClient().GetSession().ExtractValueSync().GetSession();
+
+        constexpr const char* path = "/Root/TheView";
+        constexpr const char* queryInView = "SELECT 1";
+
+        const TString creationQuery = std::format(R"(
+                CREATE VIEW `{}` WITH (security_invoker = true) AS {};
+            )",
+            path,
+            queryInView
+        );
+        ExecuteQuery(session, creationQuery);
+
+        const TString dropQuery = std::format(R"(
+                DROP VIEW IF EXISTS `{}`;
+            )",
+            path
+        );
+        ExecuteQuery(session, dropQuery);
+        // an attempt to drop an already deleted view does not produce an error
+        ExecuteQuery(session, dropQuery);
```

Comment:
We also need more tests to the following cases:
- DROP VIEW query must fail if there is no such view
- DROP VIEW query must fail if we are attempting to delete view, but it is not a view (topic, table etc)
- DROP VIEW IF EXISTS query must fail if we are attempting to delete view, but it is not a view (topic, table etc)

All these cases should work also for ExecuteQuery calls (not only for ExecuteSchemeQuery) - they are newer API

Also the same cases, but for CREATE type

Reply:
Added all 3 `DROP VIEW` tests you mentioned.

As for the `CREATE VIEW`, the test that checks that the `CREATE VIEW` statement does in fact create a view has already existed, so I haven't added it. However, I have added the other two tests for the `CREATE VIEW` statement.